tests/sentry/middleware/test_ratelimit_middleware.py

```diff
@@ -65,71 +72,61 @@ def test_fails_open(self, default_rate_limit_mock):
         request = self.factory.get("/")
         with freeze_time("2000-01-01"):
             default_rate_limit_mock.return_value = RateLimit(0, 100)
-            self.middleware.process_view(request, self._test_endpoint, [], {})
-
-    def test_process_response_fails_open(self):
-        request = self.factory.get("/")
-        bad_response = object()
-        assert self.middleware.process_response(request, bad_response) is bad_response
-
-        class BadRequest:
-            def __getattr__(self, attr):
-                raise Exception("nope")
-
-        bad_request = BadRequest()
-        assert self.middleware.process_response(bad_request, bad_response) is bad_response
+            self.middleware(request)
 
     @patch("sentry.middleware.ratelimit.get_rate_limit_value")
     def test_positive_rate_limit_check(self, default_rate_limit_mock):
         request = self.factory.get("/")
         with freeze_time("2000-01-01"):
             default_rate_limit_mock.return_value = RateLimit(0, 100)
-            self.middleware.process_view(request, self._test_endpoint, [], {})
-            assert request.will_be_rate_limited
+            resp = self.middleware(request)
```

Comment:
Without `will_be_rate_limited` to assert on I went for the headers - it doesn't feel as accurate since sometimes `X-Sentry-Rate-Limit-Remaining` is 0 but that's fine since no more requests are coming, or sometimes it's 0 and that's not fine because another request is coming and it'd be rejected.

Reply:
`will_be_rate_limited` was from a time when rate limits were not actually enforced. Now that they are, how can you determine that the request was rate limited?

Reply:
Do we actually need to know that outside of tests? It's not used anywhere else.

Reply:
> Do we actually need to know that outside of tests? 
It depends on what you mean by "that". As you pointed out, using `X-Sentry-Rate-Limit-Remaining` is not sufficient to tell us if it the request was rate limited (and `will_be_rate_limited` was). But we're testing the rate limit middleware, so presumably we should test to see that it does in fact rate limit.

> It's not used anywhere else.
Are you sure?

Reply:
Arg it's used in the access log. What I'm stuck on is that we are trying to not add things to the request object but we also want extra things on the request object like `will_be_rate_limited` and `rate_limit_category`. I tried returning 2 things from `__call__` with bad results earlier.